BlueDBOnDisk/src/main/java/io/bluedb/disk/file/FileManager.java

```diff
@@ -71,6 +71,24 @@ public void lockMoveFileUnlock(Path src, Path dst) throws BlueDbException {
 		return new BlueObjectInput<T>(readLock, serializer);
 	}
 
+	public BlueReadLock<Path> getReadLockIfFileExists(Path path) throws BlueDbException {
+		BlueReadLock<Path> lock = lockManager.acquireReadLock(path);
+		try {
+			if (exists(path)) {
+				return lock;
+			}
+		} catch (Throwable t) { // make damn sure we don't hold onto the lock
+			lock.release();
+			throw new BlueDbException("Error attempting to acquire read lock", t);
+		}
+		lock.release();
+		return null;
+	}
+
+	public boolean exists(Path path) {
+		return path.toFile().exists();

```

Comment:
I haven't mentioned things like this before and you don't need to change this. But there is a Files.exists(Path path) method and a Files.createDirectories method that allow you to use the path object instead of getting the file. Doesn't matter that much. Now that we use path objects more I like using these methods instead of the java.io.File ones

Reply:
I actually created this partly to make sure I could generate an exception and test how it's handled.